### PR TITLE
Stripe: support shipping info in purchases

### PR DESCRIPTION
**Pull request by whitby3001**
Adds support for adding a shipping address and tracking information to purchases and authorizations.

### CODE REVIEW

File: test/unit/gateways/stripe_test.rb [hunk 1]
```diff
@@ -12,6 +12,10 @@ def setup
 
     @options = {
       :billing_address => address(),
+      :shipping_address => address(),
+      :description => 'Test Purchase',
+      :carrier => 'UPS',
+      :tracking_number => '12345'
```
Comment (bpollack) [inline test/unit/gateways/stripe_test.rb]: @whitby3001 This line is missing a trailing comma, causing the test suite to fail. If you still want this, would you mind patching and rebasing on `master`? If so, I can merge this today or tomorrow.

Comment (whitby3001) [reply test/unit/gateways/stripe_test.rb]: Will do!

Comment (bpollack) [reply test/unit/gateways/stripe_test.rb]: Thanks!

**Comment by whitby3001** [comment]
Sorry, I think I've messed this up trying to do a rebase, I'll just re-fork and create a new pull request tomorrow.

**Comment by bpollack** [comment]
I can help you rebase if GitHub hasn't GC'd the original commit. I wasn't trying to make your life hard. :-)

**Comment by whitby3001** [comment]
Ok, if there's a way to fix it then I'm all for it - how do i go about that? Sorry my git-fu isn't particularly good :/

**Comment by bpollack** [comment]
Yep, `git rebase` is gross. The general way to fix a rebase gone bad is:

  1. Go back to the original commit. In your case, that's f242b8473eb9e5b8cd151dbdc4810890ed080040.
  2. Rebase from that
  3. Resolve merge conflicts

The reason your rebase got messy is because you'd merged in `master` via commit c1d6f55e5f37f8645e4429f4c2668310afd68bd4. We'll want to drop that commit in this process.

So, the process: first, while on branch `stripe-shipping-address`, you'll throw out everything except the original commit:

`git reset --hard f242b8473eb9`

Then, you'll rebase just that commits:

```
git fetch
git rebase origin/master
```

You'll have a couple merge conflicts, but they shouldn't be too bad. `git add` the files once you've resolved the conflicts. This is also probably the easiest time to add that missing comma. When you're done, `git rebase --continue` to finish the rebase.

At that point, you should have a nice linear history, with just one commit sitting beyond the current Active Merchant `master`. You'll need to force-push that branch (`git push --force`), at which point this will collapse down to a nice clean one-commit that I can easily merge in.

Let me know if you've got questions.

**Pull request closed without merging**